reuseport.go

```diff
@@ -38,19 +38,28 @@ func reuseErrShouldRetry(err error) bool {
 	}
 }
 
-// Dials using reusport and then redials normally if that fails.
-func reuseDial(ctx context.Context, laddr *net.TCPAddr, network, raddr string) (net.Conn, error) {
+// Dials using reuseport and then redials normally if that fails.
+// TODO(anacrolix): This shouldn't fail anymore: Remove fallback.
```

Comment:
This is for when we can't pick a reasonable source port. For example, when:

1. We're listening on *two* localhost addresses.
2. We're dialing a non-localhost address.

In that case, we'll use the multiDialer (we have multiple source ports to choose from) but we'll end up picking none of them.

Really, we could probably just not call `reuseDial` in this case.